src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -205,14 +205,26 @@ internal List<OpenApiServer> GetOpenApiServers(HttpRequest? httpRequest = null)
     {
         if (httpRequest is not null)
         {
-            var serverUrl = UriHelper.BuildAbsolute(httpRequest.Scheme, httpRequest.Host, httpRequest.PathBase);
+            // Handle forwarded headers directly if present
+            var scheme = httpRequest.Headers.GetCommaSeparatedValues("X-Forwarded-Proto") is [var forwardedScheme, ..]
```

Comment:
Why is this happening in the document service? We have a whole middleware for forwarded headers

Reply:
We can't always rely on the forwarded headers middleware being enabled for this feature to work --especially since the training people are used to is that OpenAPI operates independent of any other options.

There's also some local dev scenarios (Codespaces, for example) that make it hard to discover that you need to enable forwarded headers middleware.

Reply:
That applies to all of asp.net core, not this feature. There are also security concerns with blindly reading xff headers (the middleware goes through great lengths to make that configurable but secure by default). You can (as a customer) enable his flag ASPNETCORE_FORWARDEDHEADERS_ENABLED, to turn on the middleware from the outside. 

The custom should be using one of those techniques. This middleware is no more special than the rest.

Reply:
Lucky for you I don't feel argumentative today. 😆  I'll close this PR out and make the same change in the servicing PR.